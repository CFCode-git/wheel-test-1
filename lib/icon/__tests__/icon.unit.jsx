import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../icon';
import {mount} from 'enzyme'

describe('Icon.tsx', () => {
  it('render successful', () => {
    const json = renderer.create(<Icon name="setting"/>).toJSON();
    expect(json).toMatchSnapshot()
  });
  it('onClick', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="setting" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  });
});
