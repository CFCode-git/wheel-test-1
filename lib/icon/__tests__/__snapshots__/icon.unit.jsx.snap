// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.tsx render successful 1`] = `
<svg
  className="bui-icon"
>
  <use
    xlinkHref="#setting"
  />
</svg>
`;
